manta-trusted-setup/src/bin/groth16_phase2_client.rs

```diff
@@ -0,0 +1,279 @@
+use serde::{de::DeserializeOwned, Serialize};
+
+/// Welcome Message
+pub const TITLE: &str = r"
+__  __             _          _______             _           _    _____      _               
+|  \/  |           | |        |__   __|           | |         | |  / ____|    | |              
+| \  / | __ _ _ __ | |_ __ _     | |_ __ _   _ ___| |_ ___  __| | | (___   ___| |_ _   _ _ __  
+| |\/| |/ _` | '_ \| __/ _` |    | | '__| | | / __| __/ _ \/ _` |  \___ \ / _ | __| | | | '_ \ 
+| |  | | (_| | | | | || (_| |    | | |  | |_| \__ | ||  __| (_| |  ____) |  __| |_| |_| | |_) |
+|_|  |_|\__,_|_| |_|\__\__,_|    |_|_|   \__,_|___/\__\___|\__,_| |_____/ \___|\__|\__,_| .__/ 
+                                                                                        | |    
+                                                                                        |_|    
+";
+
+pub type C = Groth16BLS12381;
+pub type Config = manta_trusted_setup::ceremony::config::config::Config;
+pub type Client = manta_trusted_setup::ceremony::client::Client<C>;
+
+/// Command
+#[derive(Debug, Subcommand)]
+pub enum Command {
+    /// Register for the Trusted Setup Ceremony
+    Register,
+
+    /// Runs the Trusted Setup Ceremony as a Contributor
+    Contribute,
+}
+
+/// Command Line Arguments
+#[derive(Debug, Parser)]
+pub struct Arguments {
+    /// Command
+    #[clap(subcommand)]
+    command: Command,
+}
+
+impl Arguments {
+    /// Takes command line arguments and executes the corresponding operations.
+    #[inline]
+    pub fn run(self) -> Result<(), Error> {
+        println!("{}", TITLE);
+        match self.command {
+            Command::Register => {
+                let twitter_account = Input::with_theme(&ColorfulTheme::default())
+                    .with_prompt("Your twitter account")
+                    .interact_text()
+                    .expect("");
+                let email = Input::with_theme(&ColorfulTheme::default())
+                    .with_prompt("Your email")
+                    .interact_text()
+                    .expect("");
+                register(twitter_account, email);
+                Ok(())
+            }
+            Command::Contribute => {
+                match tokio::runtime::Builder::new_multi_thread()
+                    .worker_threads(4)
+                    .enable_io()
+                    .enable_time()
+                    .build()
+                {
+                    Ok(runtime) => runtime.block_on(async { contribute().await }),
+                    Err(err) => Err(Error::UnexpectedError(format!("{}", err))),
+                }
+            }
+        }
+    }
+}
+
+/// Prompts the client information.
+#[inline]
+pub fn prompt_client_info() -> Result<(PublicKey<C>, PrivateKey<C>), Error> {
+    println!(
+        "Please enter your {} that you get when you registered yourself using this tool.",
+        "Secret".italic()
+    );
+    let secret_str: String = Input::with_theme(&ColorfulTheme::default())
```

Comment:
@BoyuanFeng You may want to use `Password` instead of `Input` for secret: https://docs.rs/dialoguer/0.10.2/dialoguer/struct.Password.html